runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -1454,6 +1480,37 @@ protected String getKindString() {
     }
   }
 
+  private static class DataflowReadFromPubsubForRunnerV2
+      extends PTransform<PBegin, PCollection<PubsubMessage>> {
+
+    private final PubsubUnboundedSource transform;
+
+    public DataflowReadFromPubsubForRunnerV2(PubsubUnboundedSource transform) {
+      this.transform = transform;
+    }
+
+    PubsubUnboundedSource getOverriddenTransform() {
+      return this.transform;
+    }
+
+    @Override
+    public PCollection<PubsubMessage> expand(PBegin input) {
+      Coder coder = PubsubMessageProtoCoder.of();
```

Comment:
I would image the expansion here looks like: 
```java
return input.apply(new DataflowPubSubSourceRunnerv2())
.apply(MapElements() {bytes -> new PubSubMessage()})
```

Reply:
What should DataflowPubSubSourceRunnerV2 do?

Reply:
The DataflowPubSubSourceRunnerV2 is the one that `DataflowRunner` translate into `ParallelRead`.